### DISCUSSION TITLE
What does this mean and how can it be resolved properly?  Package 'Microsoft.AspNet.Mvc 5.2.8' was restored using '...' instead of the project target framework 'net8.0'. This package may not be fully compatible with your project.

### DISCUSSION THREAD

**Discussion by myyandar** [dotnet/aspnetcore#59294] [General]
What does this mean and how can it be resolved properly?

Package 'Microsoft.AspNet.Mvc 5.2.8' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework 'net8.0'. This package may not be fully compatible with your project.

EF Core version:
Database provider:  Microsoft.EntityFrameworkCore.SqlServer
Target framework:  .NET 8.0
Operating system:
IDE: Visual Studio 2022 17.12.1

This is not a coding issue.

**Comment by ErikEJ** [comment]
You have a reference to an old ASP.NET dll (.NET Framework) - please share a full repro.

What does this have to do with EF Core?

**Comment by myyandar** [comment]
The problem has been there since the very first time I updated packages. Perhaps my project file may give indications:

<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
    <UserSecretsId>aspnet-IssWeb-96ac3037-0def-4d49-b7a7-f3bd94d2fc41</UserSecretsId>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Azure.Identity" Version="1.11.4" />
    <PackageReference Include="MailKit" Version="4.8.0" />
    <PackageReference Include="Microsoft.AspNet.Mvc" Version="5.2.8" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.Facebook" Version="8.0.10" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.Google" Version="8.0.10" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.MicrosoftAccount" Version="8.0.10" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.Twitter" Version="8.0.10" />
    <PackageReference Include="Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore" Version="8.0.10" />
    <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="8.0.10" />
    <PackageReference Include="Microsoft.AspNetCore.Identity.UI" Version="8.0.10" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="8.0.10" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="8.0.10" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="8.0.10">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.IdentityModel.JsonWebTokens" Version="8.2.1" />
    <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="8.0.7" />
    <PackageReference Include="MimeKit" Version="4.8.0" />
    <PackageReference Include="PayPalCheckoutSdk.DotNet7" Version="1.0.5" />
    <PackageReference Include="Stripe.net" Version="47.1.0-beta.3" />
    <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="8.2.1" />
    <PackageReference Include="System.Text.Json" Version="9.0.0" />
  </ItemGroup>

  <ItemGroup>
    <Folder Include="Areas\Admin\Data\" />
    <Folder Include="Areas\Admin\Models\" />
  </ItemGroup>

  <ItemGroup>
    <None Include="wwwroot\Templates\EmailTemplate.html" />
  </ItemGroup>

</Project>

**Comment by myyandar** [comment]
![Image](https://github.com/user-attachments/assets/c6dee902-02a7-45f2-842a-426f808e2bc6)

**Comment by myyandar** [comment]
Please guide me through updating the "old.dll" you mentioned to a .Net 8 MVC web application-compatible one.

**Comment by ErikEJ** [comment]
You need to format you project file as code, no one can read it here.

**Comment by myyandar** [comment]
```
`<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
    <UserSecretsId>aspnet-IssWeb-96ac3037-0def-4d49-b7a7-f3bd94d2fc41</UserSecretsId>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Azure.Identity" Version="1.11.4" />
    <PackageReference Include="MailKit" Version="4.8.0" />
    <PackageReference Include="Microsoft.AspNet.Mvc" Version="5.3.0" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.Facebook" Version="8.0.10" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.Google" Version="8.0.10" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.MicrosoftAccount" Version="8.0.10" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.Twitter" Version="8.0.10" />
    <PackageReference Include="Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore" Version="8.0.10" />
    <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="8.0.10" />
    <PackageReference Include="Microsoft.AspNetCore.Identity.UI" Version="8.0.10" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="8.0.10" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="8.0.10" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="8.0.10">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.IdentityModel.JsonWebTokens" Version="8.2.1" />
    <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="8.0.7" />
    <PackageReference Include="MimeKit" Version="4.8.0" />
    <PackageReference Include="PayPalCheckoutSdk.DotNet7" Version="1.0.5" />
    <PackageReference Include="Stripe.net" Version="47.1.0-beta.3" />
    <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="8.2.1" />
    <PackageReference Include="System.Text.Json" Version="9.0.0" />
  </ItemGroup>

  <ItemGroup>
    <Folder Include="Areas\Admin\Data\" />
    <Folder Include="Areas\Admin\Models\" />
  </ItemGroup>

  <ItemGroup>
    <None Include="wwwroot\Templates\EmailTemplate.html" />
  </ItemGroup>

</Project>
`
```

**Comment by ajcvickers** [comment]
Moving to ASP.NET, since this is one of their packages and is not something that EF depends on.

**Comment by martincostello** [comment]
Why are you trying to use a package for .NET Framework (Microsoft.AspNet.Mvc) in an ASP.NET Core project?

**Comment by myyandar** [comment]
**> Why are you trying to use a package for .NET Framework (Microsoft.AspNet.Mvc) in an ASP.NET Core project?**

Thank you very much for your attention to my query. I know that queries from newbies like me can be nerve-tingling. Forgive me.

The package Microsoft.AspNet.MVC version 5.3.0 was already installed when I generated the application from the start in Visual Studio Community 2022. The first Build message was:

**1>C:\Users\myyan\source\repos\IssWeb\IssWeb\IssWeb.csproj : warning NU1701: Package 'Microsoft.AspNet.Mvc 5.3.0' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework 'net8.0'. This package may not be fully compatible with your project.**

Seeing this message, I wanted to see if a lower version would work. So, I installed version 5.2.8 of the package, but the problem persisted. 

From your last question, I can infer that the package is not a .Net or .NetCore package. Does it mean I can safely remove it from the packages?

**Comment by martincostello** [comment]
If you don't know why it is there, then I suggest removing it and seeing it the problem goes away and the project compiles successfully.

**Comment by myyandar** [comment]
Thanks always for the attention. I removed the package (Microsoft.AspNet.Mvc 5.2.8) and my app was stripped of its _Layout formatting. It now displays everything inline, no graphics.

**Comment by dotnet-policy-service** [comment]
Thank you for filing this issue. In order for us to investigate this issue, please provide [a minimal repro project](https://github.com/dotnet/aspnetcore/blob/main/docs/repro.md) that illustrates the problem without unnecessary code. Please share with us in a public GitHub repo because we cannot open ZIP attachments, and don't include any confidential content.

**Comment by wtgodbe** [comment]
This package (Microsoft.AspNet.Mvc) is not meant to be used with .NET core, it is an old framework package. Converting this to a discussion in case folks from the community want to help out with your specific issue, but in general the contents of that package should now be in the ASP.NET shared framework by default.
